plugins/inputs/logstash/logstash.go

```diff
@@ -0,0 +1,284 @@
+package logstash
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"net/url"
+	"time"
+
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/plugins/inputs"
+	jsonparser "github.com/influxdata/telegraf/plugins/parsers/json"
+)
+
+const sampleConfig = `
+  ## This plugin reads metrics exposed by Logstash Monitoring API.
+  #
+  # logstashURL = "http://localhost:9600"
+  URL = "http://localhost:9600"
+`
+const jvmStats = "/_node/stats/jvm"
+const processStats = "/_node/stats/process"
+const pipelineStats = "/_node/stats/pipeline"
+
+type Logstash struct {
+	URL    string
+	client *http.Client
+}
+
+type JVMStats struct {
+	ID  string      `json:"id"`
+	JVM interface{} `json:"jvm"`
+}
+
+type ProcessStats struct {
+	ID      string      `json:"id"`
+	Process interface{} `json:"process"`
+}
+
+type PluginEvents struct {
+	QueuePushDurationInMillis float64 `json:"queue_push_duration_in_millis"`
+	DurationInMillis          float64 `json:"duration_in_millis"`
+	In                        float64 `json:"in"`
+	Out                       float64 `json:"out"`
+}
+
+type Plugin struct {
+	ID     string       `json:"id"`
+	Events PluginEvents `json:"events"`
+	Name   string       `json:"name"`
+}
+
+type PipelinePlugins struct {
+	Inputs  []Plugin `json:"inputs"`
+	Filters []Plugin `json:"filters"`
+	Outputs []Plugin `json:"outputs"`
+}
+
+type PipelineQueue struct {
+	Events   float64     `json:"events"`
+	Qtype    string      `json:"type"`
+	Capacity interface{} `json:"capacity"`
+	Data     interface{} `json:"data"`
+}
+
+type Pipeline struct {
+	Events  interface{}     `json:"events"`
+	Plugins PipelinePlugins `json:"plugins"`
+	Reloads interface{}     `json:"reloads"`
+	Queue   PipelineQueue   `json:"queue"`
+}
+
+type PipelineStats struct {
+	ID       string   `json:"id"`
+	Pipeline Pipeline `json:"pipeline"`
+}
+
+//Description returns short info about plugin
+func (l *Logstash) Description() string { return "Read metrics exposed by Logstash" }
+
+//SampleConfig returns details how to configure plugin
+func (l *Logstash) SampleConfig() string { return sampleConfig }
+
+//createHttpClient create clients to access API
+func (l *Logstash) createHTTPClient() (*http.Client, error) {
+
+	client := &http.Client{
+		Timeout: time.Duration(4 * time.Second),
```

Comment:
Add a transport, we don't want to use the default one:
```
Transport: &http.Transport{
	Proxy: ProxyFromEnvironment,
},
```

Not required, but would be nice to support TLS too.